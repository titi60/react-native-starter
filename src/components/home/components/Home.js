import React from 'react';
import Config from 'react-native-config';

import getTranslations from 'utils/locales';

import Button from 'shared/Button';
import Box from 'shared/Box';
import Text from 'shared/Text';

const { GITHUB_TOKEN } = Config;

const Home = ({ navigation }) => (
  <Box flex={1} justifyContent="center" alignItems="center">
    <Text fontSize={4}>{getTranslations('home', 'page_title')}</Text>

    <Button onPress={() => navigation.navigate('Details')}>
      <Text mt={3}>{getTranslations('home', 'navigation_details')}</Text>
    </Button>

    {GITHUB_TOKEN !== '' && (
      <Button onPress={() => navigation.navigate('Github')}>
        <Text mt={3}>{getTranslations('home', 'navigation_github')}</Text>
      </Button>
    )}
  </Box>
);

export default Home;
