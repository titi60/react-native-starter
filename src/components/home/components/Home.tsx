import React, { useCallback, ReactElement } from 'react';

import { INavigationTypes } from 'types/navigation.types';

import getTranslations from '@utils/locales';

import Button from '@shared/Button';
import Box from '@shared/Box';
import Text from '@shared/Text';
import SafeView from '@shared/SafeView';

interface IProps {
  navigation: INavigationTypes;
}

const Home = ({ navigation }: IProps): ReactElement => {
  const goToDetails = useCallback(() => navigation.navigate('Details'), []);

  return (
    <SafeView>
      <Box flex={1} justifyContent="center" alignItems="center">
        <Text testID="home_title" fontSize={4}>
          {getTranslations('home', 'page_title')}
        </Text>

        <Button testID="goto_details" onPress={goToDetails}>
          <Text mt={3}>{getTranslations('home', 'navigation_details')}</Text>
        </Button>
      </Box>
    </SafeView>
  );
};

export default Home;
