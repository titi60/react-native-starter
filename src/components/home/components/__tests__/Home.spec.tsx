/* eslint-disable @typescript-eslint/no-explicit-any */

import React from 'react';
import { shallow } from 'enzyme';

import Home from '../Home';

describe('Home page component', () => {
  // given
  const props = { navigation: { navigate: jest.fn() } as any };

  it('should render correctly', () => {
    // When
    const wrapper = shallow(<Home {...props} />);

    // Then
    expect(wrapper).toMatchSnapshot();
  });

  it('should trigger the goBack method when the button is pressed', () => {
    // When
    const wrapper = shallow(<Home {...props} />);

    wrapper
      .findWhere((node) => node.prop('testID') === 'goto_details')
      .simulate('press');

    // Then
    expect(props.navigation.navigate.mock.calls.length).toBe(1);
  });
});
