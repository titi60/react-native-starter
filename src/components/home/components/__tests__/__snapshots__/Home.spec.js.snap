// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page component should render correctly 1`] = `
<Styled(View)
  alignItems="center"
  flex={1}
  justifyContent="center"
>
  <Text
    fontSize={4}
  >
    Page d'accueil
  </Text>
  <Button
    onPress={[Function]}
    testID="goto-details"
  >
    <Text
      mt={3}
    >
      Aller à la page détails
    </Text>
  </Button>
</Styled(View)>
`;
